packages/dev-approuter/lib/devApprouter.js

```diff
@@ -0,0 +1,147 @@
+     * @see https://github.com/ui5-community/ui5-ecosystem-showcase/tree/main/packages/dev-approuter
+     */
+    async start(extensions = []) {
+        // loads env from default-env.json
+        xsenv.loadEnv();
+
+        const config = parseConfig();
+        const cwd = process.cwd();
+
+        // lookup the CAP server root
+        let capServerConfig;
+        const capModules = await findCAPModules({ cwd });
+        if (capModules.length > 1) {
+            throw new Error(`Multiple CAP modules found. The package approuter-middlewaare-ui5-cap can only handle one CAP module as dependency.`);
+        } else if (capModules.length === 1) {
+            capServerConfig = capModules[0];
+        }
+
+        // find all UI5 modules from the CAP server root and dependencies from the approuter
+        const ui5Modules = [...(await findUI5Modules({ cwd, skipLocalApps: true }))];
+        if (capServerConfig) {
+            ui5Modules.push(...(await findUI5Modules({ cwd: capServerConfig.modulePath, skipDeps: true })))
+        }
+
+        // collect UI5 middlewares
+        const ui5Middlewares = [];
+        for await (const ui5Module of ui5Modules) {
+            let { moduleId, modulePath, mountPath } = ui5Module;
+
+            // create a patched router
+            const router = await createPatchedRouter();
+
+            // apply the UI5 middlewares to the router
+            await applyUI5Middleware(router, {
+                basePath: modulePath,
+                configPath: modulePath,
+            });
+
+            // mounting the router for the UI5 application to the CAP server
+            console.log(`Mounting ${mountPath} to UI5 app ${modulePath}`);
+
+            let middlewareMountPath
+            // define middlewareMountPath as `/_${mountPath}` if ui5 module is referenced as "dependency" in xs-dev.json or xs-app.json
+            if (config.dependencyRoutes && config.dependencyRoutes[moduleId]) {
+                // configure UI5 route
+                config.dependencyRoutes[moduleId] = configureUI5Route(moduleId, mountPath, config.dependencyRoutes[moduleId]);
+
+                middlewareMountPath = "/_" + mountPath;
+
+                // add destination for newly configured route
+                addDestination(moduleId, process.env.PORT, middlewareMountPath);
+            } else {
+                middlewareMountPath = mountPath;
+            }
+
+            // store the router for later registration
+            ui5Middlewares.push({
+                path: middlewareMountPath,
+                handler: router
+            });
+
+        }
+
+        // start CAP server on different port
+        if (capServerConfig) {
+            const { modulePath, moduleId } = capServerConfig;
+
+            // start CAP server on different port
+            const app = express();
+            const { servicesPaths } = await applyCAPMiddleware(app, { root: modulePath, cwd });
+            app.listen(4004, () => {
```

Comment:
Does an env variable make sense here? `process.env.CAP_PORT || 4004` - to be flexible with the CAP server port?

Reply:
Yep, that makes sense. Will also add that to the readme.